modules/localStorage.js

```diff
@@ -0,0 +1,100 @@
+const entryTask = document.getElementById('newTask');
+
+const storageAvailable = (type) => {
+  try {
+    const storage = window[type];
+    const x = '__storage_test__';
+    storage.setItem(x, x);
+    storage.removeItem(x);
+    return true;
+  } catch (e) {
+    return e instanceof DOMException && (
+    // everything except Firefox
+      e.code === 22
+    // Firefox
+    || e.code === 1014
+    // test name field too, because code might not be present
+    // everything except Firefox
+    || e.name === 'QuotaExceededError'
+    // Firefox
+    || e.name === 'NS_ERROR_DOM_QUOTA_REACHED');
+  }
+};
+
+export const savedata = () => {
+  const listOnStorage = JSON.parse(localStorage.getItem('toDoList')) || [];
+  if (storageAvailable('localStorage')) {
+    const listElement = {
+      description: `${entryTask.value}`,
+      completed: false,
+      index: JSON.parse(`${listOnStorage.length + 1}`),
+    };
+    listOnStorage.push(listElement);
+    localStorage.setItem('toDoList', JSON.stringify(listOnStorage));
+  }
+};
+
+export const retrivedata = () => {
+  const listOnStorage = JSON.parse(localStorage.getItem('toDoList')) || [];
+  const listDisplay = document.getElementById('uList');
+  document.querySelectorAll('.task-style').forEach((t) => t.remove());
+
+  listOnStorage.forEach((tsk) => {
+    const line = document.createElement('li');
+    line.innerHTML = `
+    <span>
+      <input type="checkbox" id= "id-${tsk.index}" ${tsk.completed ? 'checked' : ''}>
+      <input type="text" id="tx-${tsk.index}" value= "${tsk.description}" disabled>
+    </span>
+    <i class= "menu" id="${tsk.index}">__</i>`;
+    line.classList.add('task-style');
+    line.id = `li-${tsk.index}`;
+    listDisplay.appendChild(line);
+  });
```

Comment:
# optional
lets avoid writing direct html in the js file
use javascript dom to create elements and classes for cleaner code

Reply:
Sure. But what do you think If we use this like 'backtack' it's a modern way of using JS. I read about it in a reading document. I did not remember the link now, but I read that it's good for use instead of using direct DOM because DOM affects very greatly on the browser as it loads again & again for creating DOM elements as JS read code line by line.